File: leetcode_solutions/0006-zigzag-conversion/solution.py
class Solution:
    def convert(self, s: str, numRows: int) -> str:

        if numRows == 1 or numRows >= len(s):
            return s

        rows = ["" for _ in range(numRows)]
        
        i_row = 0
        direction = -1

        for ch in s:
            rows[i_row] += ch
            if i_row == numRows - 1 or i_row == 0:
                direction = -direction

            i_row += direction
           

        return ''.join(s for s in rows)

File: leetcode_solutions/0006-zigzag-conversion/solution.java
class Solution {
    public String convert(String s, int numRows) {
        if (numRows == 1){
            return s;
        }
        String[] rows = new String[numRows];
        Arrays.fill(rows, "");
        int dir = 1;
        int iRow = 0;
        for (int iString = 0; iString < s.length(); iString++){
            rows[iRow] += s.charAt(iString);
            iRow += dir;
            if (iRow == 0 || iRow == numRows - 1){
                dir = -dir;
            }
        }
        String retStr = "";
        for (String row: rows){
            retStr += row;
        }

        return retStr;
    }
}
